keycloak/keycloak-admin-ui
# Adds User Fed Group and Role mappers

## Motivation
https://issues.redhat.com/browse/APPDUX-929 and https://issues.redhat.com/browse/APPDUX-940

## Brief Description
Adds the 3 remaining mapper types to user federation ldap providers (group-ldap-mapper, role-ldap-mapper, and hardcoded-ldap-role-mapper). These required extra functionality than the others e.g. a modal to select from existing roles/clients. Also includes cypress tests for all 3 of the new mappers.

## Verification Steps
1. Go to User Fed and create a new LDAP provider if one doesn't already exist.
2. Go to the LDAP provider settings and click the Mappers tab.
3. Click Add mapper and choose group-ldap-mapper and verify that you can create a new one successfully.
4. Repeat for role-ldap-mapper and hardcoded-ldap-role-mapper. 
5. When configuring the role mappers, verify that you can select a role via the "Select role" button (hardcoded-ldap-role-mapper) or a client from the Client ID dropdown (role-ldap-mapper).
6. Bonus points - verify that the tests for user fed mappers all complete without errors.

## Checklist:

## Additional Notes
None
